src/Catel.MVVM/Collections/FastObservableDictionary.cs

```diff
@@ -0,0 +1,1060 @@
+
+
+            if (copyList != null)
+            {
+                WriteMultipleRemoveSuspension(copyList, 0, copyList.Count);
+            }
+        }
+
+
+        #endregion
+
+        #region ICollection<KeyValuePair<TKey, TValue>>
+        public int Count => _list.Count;
+        public bool IsReadOnly { get; private set; }
+
+        public void Add(KeyValuePair<TKey, TValue> item)
+        {
+            InsertSingleValue(item.Key, item.Value, true);
+        }
+
+        public void Clear()
+        {
+            RemoveAllItems();
+        }
+
+        public bool Contains(KeyValuePair<TKey, TValue> item)
+        {
+            return ContainsKey(item.Key);
+        }
+
+        public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+        {
+            if (array.Length - arrayIndex < Count) throw new IndexOutOfRangeException("Array doesn't have enough space to copy all the elements");
+            for (var i = 0; i < Count; i++)
+            {
+                var key = _list[i];
+                var value = _dict[key];
+                array[arrayIndex + i] = new KeyValuePair<TKey, TValue>(key, value);
+            }
+        }
+
+        public bool Remove(KeyValuePair<TKey, TValue> item)
+        {
+            return TryRemoveSingleValue(item.Key, out _);
+        }
+        #endregion
+
+        #region IDictionary<TKey,TValue>
+        public TValue this[TKey key]
+        {
+            get => _dict[key];
+            set => InsertSingleValue(key, value, true);
+        }
+
+        public ICollection<TKey> Keys => _dict.Keys;
+        public ICollection<TValue> Values => _dict.Values;
+
+        public bool ContainsKey(TKey key)
+        {
+            return _dict.ContainsKey(key);
+        }
+
+        public void Add(TKey key, TValue value)
+        {
+            InsertSingleValue(key, value, true);
+        }
+
+        public bool Remove(TKey keyToRemove)
+        {
+            return TryRemoveSingleValue(keyToRemove, out _);
+        }
+
+        /// <summary>
+        /// Attempts to get the value from a key
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            return _dict.TryGetValue(key, out value);
+        }
+
+
+        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+        {
+            return AsEnumerable().GetEnumerator();
+        }
+        #endregion
+
+        #region IReadOnlyDictionary<TKey,TValue>
+        IEnumerable<TKey> IReadOnlyDictionary<TKey, TValue>.Keys => _dict.Keys;
+        IEnumerable<TValue> IReadOnlyDictionary<TKey, TValue>.Values => _dict.Values;
+        #endregion
+
+        #region IDictionary
+        ICollection IDictionary.Keys => _dict.Keys;
+        ICollection IDictionary.Values => _dict.Values;
+
+        public bool IsFixedSize => ((IDictionary)_dict).IsFixedSize;
```

Comment:
Why the need for casting? Is this property implemented explicitly on the Dictionary<TKey, TValue>?

Reply:
yes it's explicit